doc/debugger.rst

```diff
@@ -1388,6 +1402,27 @@ reestablish the link).
 ``msg`` is an optional string elaborating on the reason for the detach.  It may
 or may not be present depending on the nature of detachment.
 
+AppNotify notification (0x07)
+-----------------------------
+
+Format::
+
+    NFY <int: 0x07> [<tval>]* EOM
+
+Example::
+
+    NFY 7 "DebugPrint" "Everything is going according to plan!" EOM
+
+This is a custom notification message whose meaning and semantics depend on the
+application.
+
+AppNotify messages are used for direct communication between the debug client
+and debug target over the Duktape debug protocol. Both the meaning of a custom
+message and the dvalues it contains are entirely up to the implementation and
+depending on the needs of the application, need not be supported at all.
+
```

Comment:
More discussion on the issue itself - but I'd prefer recommending a convention to start with a string identifier (for both AppNotify and AppRequest). It will make it easier to mix debug clients and debug targets - but it naturally doesn't need to be an enforced convention.